components/process/process.go

```diff
@@ -41,7 +41,7 @@ func (provider *Provider) Start(ctx context.Context, input *core.StartInput) (*c
 }
 
 func (provider *Provider) start(ctx context.Context, componentID string, inputSpec string) (State, error) {
-	var spec Spec
+	spec := Spec{ShutdownGracePeriodSeconds: 5}
```

Comment:
This seems really subtle! I was confused at first how this was even working (assuming it is).

Seems like this will default the field when `UnmarshalString` runs only by virtue of the fact that if the field is missing, then it won't get overridden! It's not obvious to me that it wouldn't potentially be overwritten to 0.

A much clearer thing to do would be to change the field to a pointer type and then explicitly handle defaulting when the timeout is nil.

Reply:
Discussed in person but it seems reasonably common. It's first answer in both of the top two results on Google for "go json unmarshal default".
https://eli.thegreenplace.net/2020/optional-json-fields-in-go/#setting-default-values
https://stackoverflow.com/a/30445480